docs/adr/0009_versioning_strategy.md

```diff
@@ -0,0 +1,299 @@
+* Hybrid: per-component `VERSION` files for working state, plus a generated `/VERSIONS.yaml` for published snapshots.
+
+### Branching options
+
+* Option A — Persistent component-scoped release branches: `releases/<component>/v<major>.<minor>` maintained as long-lived maintenance lines.
+* Option B — Tag-only / short-lived release PRs: create ephemeral release-candidate branches or push tags directly; do not maintain long-lived release branches.
+
+### Compatibility matrix (high-level)
+
+| Versioning \\\ Branching | Option A (persistent branches) | Option B (tag-only / short-lived) |
+| ----------------------: | :----------------------------: | :-------------------------------: |
+| Per-component `VERSION`  | Good — native fit: branches host `VERSION`, easy hotfixes | OK — works, but hotfixes require creating temporary branches; more manual steps |
+| Central `VERSIONS.yaml`  | OK — CI must reconcile branch-local changes into central file; potential conflicts | Fair — central file updated from tags only; high merge-conflict risk if many concurrent releases |
+| Hybrid (chosen)         | Best — combines autonomy and a single snapshot; CI can generate `/VERSIONS.yaml` on publish | Good — possible but increases CI complexity to update central snapshot safely |
+
+### Rejection rationale (short)
+
+* Central-only `VERSIONS.yaml` + tag-only branching: rejected because hotfix/backport workflows become cumbersome and the central file becomes a high-conflict surface.
+* Per-component `VERSION` without central snapshot: rejected because consumers (website, docs) need a single reproducible snapshot; lacking that reduces discoverability and reproducibility of root releases.
+
+## Operational Rules and Rationale
+
+### Authoritative sources
+
+* Working state (pre-publication): `ocm-<component>/VERSION` located on the appropriate branch (either `main` for ongoing development, or a `releases/...` branch for maintenance lines).
+* Published state: annotated Git tag `<component>/v<major>.<minor>.<patch>` — tags are authoritative for published artifacts and their provenance.
+* Global snapshot: `/VERSIONS.yaml` — generated by CI after successful publishes and updated via OCM bot PRs; **this file is not intended for manual edits**.
+
+Rationale: Having both per-component `VERSION` files (working state) and a generated `/VERSIONS.yaml` provides autonomy and a single snapshot for consumers.
+
+### Branching model
+
+* Create and maintain persistent branches for each active minor per component using the naming pattern: `releases/<component>/v<major>.<minor>`.
+* `main` remains the integration branch for new features and non-release work.
+* Release preparation: prepare a release branch from `main` when a new minor version should be released.
+* Hotfixes: open PRs targeting the applicable `releases/<component>/vX.Y` branch; merge triggers a patch release from that branch.
+
+### Special Case: OCM Root Component (No Release Branches)
```

Comment:
I disagree with a few statements here:

1. Why would we not have OCM-specific hotfixes? If someone wants to ship OCM, we would ship both controllers and CLI, and not just one right? that means hotfixes have to be backported just the same
2. What use do we have for a snapshot-only component? We need to do conformance testing on this version, so IMO its important to also release this version and make it our single source of truth for availability.

Reply:
1. Having the submodules released should bump the ocm version automatically. Having the edge-case of requiring multiple submodules together, would be handled manually, by flagging the releases of the submodules as without bumping the OCM root component. The bump will happen via another workflow. I'm not sure if such - hopefully rare - use cases can be coordinated automatically, therefore kept them with flags/manual workflows. So, we're able to create hotfixes for OCM, but by hot fixing any of the submodules

2. I agree that testing is required, but why does this make an own release of ocm necessary and what is the benefit of introducing releases and related branches for ocm?

Reply:
For me separate testing is necessary for conformance and so that not every part of ocm has to integrate conformance for everything. Instead we can choose to have signals on top level that only push down alarms for subcomponents if they fail. See the kubernetes test matrix as an example

Reply:
sounds resonsonable. Let me think this through and come up with a better proposal :-)